src/js/starter_app.js

```diff
@@ -0,0 +1,93 @@
+/*jshint esversion: 6 */
+import Util from './util';
+import Patient from './patient';
+import FHIRClient from '../../node_modules/fhirclient/fhir-client.js';
+import $ from '../../node_modules/jquery/src/jquery';
+
+class StarterApp {
+  static extractData() {
+   
+    const ret = $.Deferred();     
+    
+    function onError() {
+      console.log('Loading error', arguments);
+      ret.reject();
+    }     
+
+    function onReady(smart)  {
+      if (smart.hasOwnProperty('patient')) { 
+        const patient = smart.patient;
+        const pt = patient.read();
+        const obv = smart.patient.api.fetchAll({
+                      type: 'Observation', 
+                      query: {
+                        code: {
+                          $or: ['http://loinc.org|8302-2', 'http://loinc.org|8462-4',
+                                'http://loinc.org|8480-6', 'http://loinc.org|2085-9',
+                                'http://loinc.org|2089-1']
+                              }
+                             }
+                    });
+        
+        $.when(pt, obv).fail(onError);
+
+        $.when(pt, obv).done(function(patient, obv) {
+          const byCodes = smart.byCodes(obv, 'code');
+          const gender = patient.gender;
+          const dob = new Date(patient.birthDate);     
+          const day = dob.getDate(); 
+          const monthIndex = dob.getMonth() + 1;
+          const year = dob.getFullYear();
+
+          const dobStr = monthIndex + '/' + day + '/' + year;
+          
+          const fname = patient.name[0].given.join(' ');
```

Comment:
we should check to see if patient.name exists first and then do the array subscript after